app/actions/remote/custom_profile.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {forceLogoutIfNecessary} from '@actions/remote/session';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@managers/network_manager';
+import {getFullErrorMessage} from '@utils/errors';
+import {logDebug} from '@utils/log';
+
+import type {CustomProfileField, CustomAttribute, CustomAttributeSet, CustomProfileAttribute, UserCustomProfileAttributeSimple} from '@typings/api/custom_profile_attributes';
+
+/**
+ * Fetches custom profile fields from the server
+ * @param serverUrl - The server URL
+ * @returns Promise with the custom profile fields or error
+ */
+export const fetchCustomProfileFields = async (serverUrl: string): Promise<{fields?: CustomProfileField[]; error?: unknown}> => {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const fields = await client.getCustomProfileAttributeFields();
+
+        if (fields?.length > 0) {
+            // Store fields in the database
+            await operator.handleCustomProfileFields({
+                fields,
+                prepareRecordsOnly: false,
+            });
+            return {fields, error: undefined};
+        }
+        return {fields: [], error: undefined};
+    } catch (error) {
+        logDebug('error on fetchCustomProfileFields', getFullErrorMessage(error));
+        forceLogoutIfNecessary(serverUrl, error);
+        return {fields: [], error};
+    }
+};
+
+/**
+ * Fetches custom profile attribute values for a user
+ * @param serverUrl - The server URL
+ * @param userId - The user ID
+ * @param filterEmpty - Whether to filter out empty values
+ * @returns Promise with the custom profile attributes or error
+ */
+export const fetchCustomProfileAttributes = async (serverUrl: string, userId: string, filterEmpty = false): Promise<{attributes: CustomAttributeSet; error: unknown}> => {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const [fields, attrValues] = await Promise.all([
+            client.getCustomProfileAttributeFields(),
+            client.getCustomProfileAttributeValues(userId),
```

Comment:
same as above, this should have a way to fetch only what has changed since a moment in time. 5/5